### DISCUSSION TITLE
Static type checking issue `reportPrivateImportUsage`

### DISCUSSION THREAD

**Discussion by ChenglongWang** [Project-MONAI/MONAI#4356] [Q&A] [answered]
When I using pyright to perform static type checking of my code, a bunch of `reportPrivateImportUsage` warning appeared. 
Here is a simple import code: 
```
from monai.config import DtypeLike, KeysCollection   # <- warning
from monai.config.type_definitions import NdarrayOrTensor  # <- no warning
```
I'm wondering which import style should I use. Is there any advise for this issue?
1) Ignore `reportPrivateImportUsage` warning
2) import from definite file
3) add explicit exports `__all__` in `__init__.py`

**Comment by Nic-Ma** [comment, marked as answer]
Hi @ChenglongWang ,

We don't use `pyright` for type check, we use `flake8`, `mypy`.
@ericspod @wyli Do you have any suggestions about this question?

Thanks.

**Reply by ericspod** [reply]
We use `flake8`, `mypy`, and `pytype` for checks. We had a discussion about what sort of import style we should follow since we're not actually consistent anywhere. I would suggest that we should change things over to be importing from the definite file when that file is meant to be publicly accessible. I don't know if we have any private files yet who's contents are used internally, they should be prefixed with underscore if we do have them. I would suggest you follow the guidance for your own code and import from the definite file, we'll look into what to do for MONAI as a whole.
